From 2e5f7b07375b5d9875f23a60e554c5bc593d8fb0 Mon Sep 17 00:00:00 2001
From: Wenzong Fan <wenzong.fan@windriver.com>
Date: Tue, 2 Jan 2018 02:56:08 +0000
Subject: [PATCH] postfix: Silence cap_dac_read_search denials

Fix avc denial:
  avc: denied { dac_read_search } for pid=630 \
  comm="master" capability=2 \
  scontext=system_u:system_r:postfix_master_t \
  tcontext=system_u:system_r:postfix_master_t tclass=capability

Upstream-Status: Backport
[https://github.com/TresysTechnology/refpolicy-contrib/commit/28ba5d0cccf057cf12b0af1e10d8244e7da23cbf]

Signed-off-by: Wenzong Fan <wenzong.fan@windriver.com>
---
 policy/modules/contrib/postfix.te | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/policy/modules/contrib/postfix.te b/policy/modules/contrib/postfix.te
index 5030567..9e1856f 100644
--- a/policy/modules/contrib/postfix.te
+++ b/policy/modules/contrib/postfix.te
@@ -171,7 +171,7 @@ optional_policy(`
 # Common postfix server domain local policy
 #
 
-allow postfix_server_domain self:capability { setuid setgid dac_override };
+allow postfix_server_domain self:capability { dac_read_search setuid setgid dac_override };
 
 allow postfix_server_domain postfix_master_t:unix_stream_socket { connectto rw_stream_socket_perms };
 
@@ -189,7 +189,7 @@ corenet_tcp_sendrecv_all_ports(postfix_server_domain)
 # Common postfix user domain local policy
 #
 
-allow postfix_user_domains self:capability dac_override;
+allow postfix_user_domains self:capability { dac_read_search dac_override };
 
 domain_use_interactive_fds(postfix_user_domains)
 
@@ -198,7 +198,7 @@ domain_use_interactive_fds(postfix_user_domains)
 # Master local policy
 #
 
-allow postfix_master_t self:capability { chown dac_override kill fowner setgid setuid sys_tty_config };
+allow postfix_master_t self:capability { chown dac_read_search dac_override kill fowner setgid setuid sys_tty_config };
 allow postfix_master_t self:capability2 block_suspend;
 allow postfix_master_t self:process { setrlimit getsched };
 allow postfix_master_t self:tcp_socket create_stream_socket_perms;
@@ -481,7 +481,7 @@ optional_policy(`
 # Map local policy
 #
 
-allow postfix_map_t self:capability { dac_override setgid setuid };
+allow postfix_map_t self:capability { dac_read_search dac_override setgid setuid };
 allow postfix_map_t self:tcp_socket { accept listen };
 
 allow postfix_map_t postfix_etc_t:dir manage_dir_perms;
-- 
2.13.3

